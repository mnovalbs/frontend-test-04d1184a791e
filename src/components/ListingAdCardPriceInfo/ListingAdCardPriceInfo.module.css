.priceInfo {
  display: flex;
  flex-direction: column;
  gap: 4px;
  text-align: right;
}

.priceRange {
  font-weight: 600;
  font-size: 19px;
  line-height: 26px;
}

.subPriceLabel {
  color: #787d9c;
  font-weight: 400;
  font-size: 14px;
  line-height: 19px;
}

@media screen and (max-width: 768px) {
  .priceInfo {
    text-align: left;
    flex-direction: row;
    align-items: center;
    gap: 12px;
  }

  .priceRange {
    font-size: 1rem;
    line-height: 22px;
  }

  .subPriceLabel {
    font-size: 12px;
    line-height: 16px;
  }
}