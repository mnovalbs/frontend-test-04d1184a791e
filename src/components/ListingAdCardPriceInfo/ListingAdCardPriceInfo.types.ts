import { Ad } from "types/Ad";

export interface ListingAdCardPriceInfoProps {
  psfMin: Ad["psf_min"];
  psfMax: Ad["psf_max"];
  subPriceLabel: Ad["subprice_label"];
}
