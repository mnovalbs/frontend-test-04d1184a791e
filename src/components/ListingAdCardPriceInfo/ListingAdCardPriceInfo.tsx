import { currencyFormatter } from "utils/formatter";
import styles from "./ListingAdCardPriceInfo.module.css";

import { ListingAdCardPriceInfoProps } from "./ListingAdCardPriceInfo.types";

function ListingAdCardPriceInfo({
  psfMin,
  psfMax,
  subPriceLabel,
}: ListingAdCardPriceInfoProps) {
  return (
    <div className={styles.priceInfo}>
      <div className={styles.priceRange}>
        {currencyFormatter(psfMin)} - {currencyFormatter(psfMax)} psf
      </div>
      <div className={styles.subPriceLabel}>{subPriceLabel}</div>
    </div>
  );
}

export default ListingAdCardPriceInfo;
