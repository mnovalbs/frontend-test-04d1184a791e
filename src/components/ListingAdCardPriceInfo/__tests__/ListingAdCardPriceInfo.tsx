import { render } from "@testing-library/react";
import mockAd from "contants/mockAd";
import ListingAdCardPriceInfo from "../ListingAdCardPriceInfo";

describe("ListingAdCardPriceInfo", () => {
  it("renders price info", () => {
    const { container } = render(
      <ListingAdCardPriceInfo
        psfMin={mockAd.psf_min}
        psfMax={mockAd.psf_max}
        subPriceLabel={mockAd.subprice_label}
      />
    );

    expect(container).toMatchSnapshot()
  });
});
