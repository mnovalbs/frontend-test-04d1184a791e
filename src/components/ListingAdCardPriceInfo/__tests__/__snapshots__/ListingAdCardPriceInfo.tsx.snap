// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingAdCardPriceInfo renders price info 1`] = `
<div>
  <div
    class="priceInfo"
  >
    <div
      class="priceRange"
    >
      $2,609
       - 
      $3,043
       psf
    </div>
    <div
      class="subPriceLabel"
    >
      Prices from $1.99M
    </div>
  </div>
</div>
`;
